File: C++/프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <bits/stdc++.h>

using namespace std;

bool compare(string a, string b) {
    return a + b > b + a;
}
string solution(vector<int> numbers) {
    string answer = "";
    bool isZero = true;
    vector<string> s;
    
    for(int n : numbers) {
        s.push_back(to_string(n));
        if(n != 0) {
            isZero = false;
        }
    }
    if(isZero) {
        return "0";
    }
    
    sort(s.begin(), s.end(), compare);
    for(string st: s) {
        answer += st;
    }
    return answer;
}

File: Java/프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        StringBuilder sb = new StringBuilder();
        Integer[] boxed = Arrays.stream(numbers).boxed().toArray(Integer[]::new);
        Arrays.sort(boxed, (a, b) -> (b.toString() + a.toString()).compareTo(a.toString() + b.toString()));
        for(int num : boxed) {
            sb.append(num);
        }
        
        if(sb.charAt(0) == '0') {
            return "0";
        }
        return sb.toString();
    }
}